views/user/authorized_index.ejs

```diff
@@ -4364,6 +4375,14 @@
                         }
                     }
                 });
+                //sets the zoom and pan limits for the individual class graphs
+                charts[id+1].options.plugins.zoom.limits = {
+                    xAxis: {min: minDate, max: maxDate},
```

Comment:
Do you define `minDate` and `maxDate` anywhere in this function?

Reply:
Those lines are in a different function and the variables appear to be implicitly declared, which is not good practice. If you intend for them to be global variables, you should declare them in the global scope.

Reply:
Those lines are in a different function and the variables appear to be implicitly declared, which is not good practice. If you intend for them to be global variables, you should declare them in the global scope.